### PR TITLE
[cxx-interop] Make sure interop does not trigger TBD validation errors

### PR DESCRIPTION
**Pull request by egorzhdan**
rdar://83405989 / https://github.com/apple/swift/issues/56458

### CODE REVIEW

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please test

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please test

**Comment by hyp** [comment]
I think that's fine, but we should be mindful that it's not necessarily a fix, rather a workaround. We do have additional issues tracking this here:
https://github.com/apple/swift/issues/61453, but there are probably other cases.

File: lib/Frontend/ArgsToFrontendOptionsConverter.cpp [hunk 1]
```diff
@@ -410,6 +411,14 @@ void ArgsToFrontendOptionsConverter::computeTBDOptions() {
       Diags.diagnose(SourceLoc(), diag::error_unsupported_option_argument,
                      A->getOption().getPrefixedName(), value);
     }
+  } else if (Args.hasArg(OPT_enable_experimental_cxx_interop) ||
+             Args.hasArg(OPT_cxx_interoperability_mode)) {
+    // TBD validation currently emits erroneous diagnostics when C++ interop is
```
Comment (hyp) [inline lib/Frontend/ArgsToFrontendOptionsConverter.cpp:416]: I would argue that the diagnostics are **not** erroneous, but rather that IRGen is incorrectly applying linkage and/or visibility to the symbols. I think we need to clarify in the comment that this is specifically a workaround for **correct** diagnostics, and that it also must be a ``FIXME``.

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

File: lib/Frontend/ArgsToFrontendOptionsConverter.cpp (hunk 1 above)
Comment (egorzhdan) [reply lib/Frontend/ArgsToFrontendOptionsConverter.cpp]: Thanks, that makes sense, I updated the comment.

**Pull request merged**